AsyncDisplayKit/Details/ASDataController.mm

```diff
@@ -651,13 +665,18 @@ - (NSIndexPath *)indexPathForNode:(ASCellNode *)cellNode;
 - (NSArray *)nodesAtIndexPaths:(NSArray *)indexPaths
 {
   ASDisplayNodeAssertMainThread();
-  return ASFindElementsInMultidimensionalArrayAtIndexPaths(_completedNodes, [indexPaths sortedArrayUsingSelector:@selector(compare:)]);
+  return ASFindElementsInMultidimensionalArrayAtIndexPaths([self externalCompletedNodes], [indexPaths sortedArrayUsingSelector:@selector(compare:)]);
 }
 
 - (NSArray *)completedNodes
 {
   ASDisplayNodeAssertMainThread();
-  return _completedNodes;
+  return [self externalCompletedNodes];
+}
+
+- (NSMutableArray *)externalCompletedNodes
+{
+  return _externalCompletedNodes != nil ? _externalCompletedNodes : _completedNodes;
```

Comment:
Yeah, the main thing is `-nodesAtIndexPaths:` needs a mutable array ([here](https://github.com/facebook/AsyncDisplayKit/pull/688/files#diff-c03b807fdd44839fd88691746d9dcc27R668)). Maybe casting it is fine?

> Not sure if having the header show NSArray as the return type will work seamlessly if the implementation actually returns NSMutableArray.

I either get a NSArray (and still need to cast), or a "duplicate declaration of method completedNodes" error :(